nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/SplitText.java

```diff
@@ -288,7 +290,7 @@ public void process(InputStream in) throws IOException {
         int fragmentIndex = 1; // set to 1 to preserve the existing behavior *only*. Perhaps should be deprecated to follow the 0,1,2... scheme
         String fragmentId = UUID.randomUUID().toString();
 
-        if (computedSplitsInfo.size() == 0) {
+        if ((computedSplitsInfo.size() == 0) && (headerFlowFile != null)) {
```

Comment:
I assume the original author had a reason, the description of the processor says "If after computation of the header there are no more data, the resulting split will consists of only header lines."